Add README for Spring 2021 JS Chats week 9: TypeScript (#221)
uclaacm/hack.uclaacm.com: posts/spring2021/js-chats-7-typescript/index.md

	return elements[elements.length - 1];
}

// Explicitly passing generics
let num = lastOf<number>([1, 6, 3, 5]);

// TypeScript infers that this variable is a string.
let str = lastOf(["an", "array", "of", "strings"]);

str = 2 // TypeScript error, str is a string
num = "Not a number" // TypeScript error, num is a number
```

## Library Support

npm libraries largely fall under three categories:

1. Packages with types defined internally
2. Popular packages without defined types, but with community supplied types
3. Not-so-popular packages without defined types or community supplied types

With the first class of npm packages, no changes need to be made. However, for the second class of packages, there is a [community-supported monorepo](https://github.com/DefinitelyTyped/DefinitelyTyped) from which types can be seamlessly added. For instance, to install the types for *lodash* (a package that doesn't provide types) as a development dependency, we can run the following command:

```tsx
npm install --save-dev @types/lodash
```

All types and TypeScript related packages should be installed as development dependencies, as they only used in development and not in production builds.

> For the third class of packages, we can define types based on the package's documentation. [Here](https://www.credera.com/insights/typescript-adding-custom-type-definitions-for-existing-libraries)'s a useful resource for those looking to write custom types for a package.

Reviewer: nit: It's usually better to include some information on what the link is in the prose itself:

```suggestion
For the third class of packages, we can define types based on the package's documentation. [Credera](https://www.credera.com/insights/typescript-adding-custom-type-definitions-for-existing-libraries) has some useful resources for those looking to write custom types for a package.
```
